src/com/etendoerp/metadata/builders/ToolbarBuilder.java

```diff
@@ -58,15 +62,28 @@ private JSONObject buildToolbarJSON(Window window) throws Exception {
             buttons.put(createButtonJSON(config));
         }
 
-        for (Tab tab : window.getADTabList()) {
-            JSONArray processButtons = getProcessButtons(tab);
-            for (int i = 0; i < processButtons.length(); i++) {
-                buttons.put(processButtons.get(i));
+        if (tabId != null) {
+            Tab tab = OBDal.getInstance().get(Tab.class, tabId);
+            if (tab != null) {
+                JSONArray processButtons = getProcessButtons(tab);
+                for (int i = 0; i < processButtons.length(); i++) { // Fixed loop condition
+                    buttons.put(processButtons.get(i));
+                }
+            }
+        } else {
+            for (Tab tab : window.getADTabList()) {
+                if (tab.isActive()) {
+                    JSONArray processButtons = getProcessButtons(tab);
+                    for (int i = 0; i < processButtons.length(); i++) {
+                        buttons.put(processButtons.get(i));
```

Comment:
JSONArray processButtons = getProcessButtons(tab);
                    for (int i = 0; i < processButtons.length(); i++) {
                        buttons.put(processButtons.get(i));
                        
                        se repite dos veces, quizas se puede extraer a un metodo?